### PR TITLE
AB#292 Feature/textfield component

### PR DESCRIPTION
**Pull request by mmexvr**
**Describe pull-request**  
_Describe what the pull-request is about_

- Added textfield component
- Added variables for textfield component
- Added story for textfield
- Improved build time, for netlify! Can be improved even more

Storybook settings
- Added story for divider
- Added background, viewport  and grid settings for storybook

- Remove some files

Next step:

TODO:
- Helper text
- Suffix
- Prefix
- Error
- Textcounter

**Solving issue**  
_Add which issue this pull-request solves by adding # plus the number of the issue (for example #123)_
Fixes: [AB#292](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/292)

**How to test**  
_Add description how to test if possible_
- Run storybook/netlify
- Possible to use regular html for some of the component parts
- Label inside needs to be a web-component
- Check textfields

**Additional context**  
_Add any other context about the pull-request here._
is it okey to supply regular html AND web-component?

### CODE REVIEW

**Comment by niklasros** [comment]
On white background, the text fields has the wrong background colour. Should be $grey-50.

text-fields-on-white

**Comment by niklasros** [comment]
Label inside: Would be great if the text could have a transition when going from default to focused where it moves to its new position while decreasing in size, in stead of having the 'hard switch' that we now have. Similiar to how Material are doing it.

**Comment by mmexvr** [comment]
> On white background, the text fields has the wrong background colour. Should be $grey-50.
> 
> text-fields-on-white

#68 Added issue for the theming

File: components/src/components/textfield/textfield.stories.js [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  placeholderText: 'Placeholder'
+};
+
+
+const textfieldDisabledTemplate = ({placeholderText}) => {
+  return `
+  ${style}
+  <c-theme name="scania" global="true"></c-theme>
+  <div class="demo">
+    Regular HTML
```
[the remark this answers is not in the document]
Comment (mmexvr) [reply components/src/components/textfield/textfield.stories.js:96]: Fixed

[the remark this answers is not in the document]
Comment (mmexvr) [reply components/src/components/textfield/textfield.stories.js:96]: Removed global

File: components/.storybook/preview.js [hunk 2]
```diff
@@ -4,10 +4,69 @@ import customElements from '../dist/collection/custom-elements.json';
...
+    name: 'Small',
+    styles: {
+      width: '320px',
+      height: '963px',
+    },
+  },
+  Medium: {
+    name: 'Medium',
+    styles: {
+      width: '627px',
```
[the remark this answers is not in the document]
Comment (mmexvr) [reply components/.storybook/preview.js]: Fixed viewport

**Review by niklasros — APPROVED** [review]
Looks good! Guess the previous comment about the label-inside transition can be done during the next build of the text field!

https://github.com/scania-digital-design-system/sdds/pull/67#issuecomment-772435451

**Review by helloimela — APPROVED** [review]

**Comment by mmexvr** [comment]
#75

**Pull request merged**